jupyter_client/kernelspec.py

```diff
@@ -54,19 +56,43 @@ def to_json(self):
         """
         return json.dumps(self.to_dict())
 
+
+_kernel_name_pat = re.compile(r'^[a-z0-9._\-]+$', re.IGNORECASE)
+
+def _is_valid_kernel_name(name):
+    """Check that a kernel name is valid."""
+    # quote is not unicode-safe on Python 2
+    return _kernel_name_pat.match(name)
+
+
+_kernel_name_description = "Kernel names can only contain ASCII letters and numbers and these separators: -._"
```

Comment:
Perhaps to make visually more obvious for users, add a space between separators. `- . _` and refer to the separators by name `...these separators (hyphen, period, underscore): - . _`